phoenix-core/src/main/java/org/apache/phoenix/calcite/PhoenixSchema.java

```diff
@@ -329,24 +333,14 @@ private TableRef fixTableMultiTenancy(TableRef tableRef) throws SQLException {
     
     private PhoenixSequence resolveSequence(String name) {
         try {
-            // FIXME: Do this the same way as resolving a table after PHOENIX-2489.
-            String tenantId = pc.getTenantId() == null ? null : pc.getTenantId().getString();
-            String q = "select 1 from " + PhoenixDatabaseMetaData.SYSTEM_SEQUENCE
-                    + " where " + PhoenixDatabaseMetaData.SEQUENCE_SCHEMA
-                    + (schemaName == null ? " is null" : " = '" + schemaName + "'")
-                    + " and " + PhoenixDatabaseMetaData.SEQUENCE_NAME
-                    + " = '" + name + "'"
-                    + " and " + PhoenixDatabaseMetaData.TENANT_ID
-                    + (tenantId == null ? " is null" : " = '" + tenantId + "'");
-            ResultSet rs = pc.createStatement().executeQuery(q);
-            if (rs.next()) {
-                return new PhoenixSequence(schemaName, name, pc);
-            }
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
+            SequenceManager manager = new SequenceManager((PhoenixStatement)pc.createStatement());
+            manager.newSequenceReference(pc.getTenantId(), TableName.createNormalized(schemaName, name) , null, SequenceValueParseNode.Op.NEXT_VALUE);
```

Comment:
Good question. From my testing, passing in null doesn't work. A static member is an option if nothing else.